### DISCUSSION TITLE
Installing CONNAISSEUR 2.0 fails

### DISCUSSION THREAD

**Discussion by apopaa** [sse-secure-systems/connaisseur#198] [Q&A] [answered]
UPDATE: Please ignore, I overlooked the documented requirement to run the following command in the case of an upgrade from 1.5 to 2.0 (I would have deleted this question, but I cannot):
_kubectl delete customresourcedefinitions imagepolicies.connaisseur.policy_

I had Connaisseur 1.5 installed and wanted to install the new version 2.0, but it's failing. Here's what I did.

Uninstall Connaisseur 1.5
_kubectl delete all,mutatingwebhookconfigurations,clusterroles,clusterrolebindings,configmaps,imagepolicies,secrets,serviceaccounts,crds -lapp.kubernetes.io/instance=connaisseur
kubectl delete ns connaisseur_

Delete 1.5 "connaisseur" directory:
_rm -rf ./connaisseur_

Clone Connaisseur 2.0:
_git clone https://github.com/sse-secure-systems/connaisseur.git_

Install Connaisseur 2.0:
_helm install connaisseur helm --atomic --create-namespace --namespace connaisseur_

Result:
_Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(ImagePolicy.spec.rules[0]): missing required field "verify" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[1]): unknown field "validator" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[1]): unknown field "with" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[1]): missing required field "verify" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[2]): unknown field "validator" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[2]): missing required field "verify" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[3]): unknown field "validator" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[3]): unknown field "with" in policy.connaisseur.v1.ImagePolicy.spec.rules, ValidationError(ImagePolicy.spec.rules[3]): missing required field "verify" in policy.connaisseur.v1.ImagePolicy.spec.rules]_

**Comment by apopaa** [comment, marked as answer]
For upgrade from 1.5 to 2.0 have to run:
_kubectl delete customresourcedefinitions imagepolicies.connaisseur.policy_

**Reply by xopham** [reply]
You are right. We tried to point that out at several places and fixed it for upcoming releases.
Sorry for the inconvenience. There seemed no simple way back.

**Comment by apopaa** [comment]
Notice that even though I included cdrs in
_kubectl delete all,mutatingwebhookconfigurations,clusterroles,clusterrolebindings,configmaps,imagepolicies,secrets,serviceaccounts,**crds** -lapp.kubernetes.io/instance=connaisseur_

I still had to run
_kubectl delete customresourcedefinitions imagepolicies.connaisseur.policy_

I'm guessing that "imagepolicies.connaisseur.policy" CRD is not labelled with "app.kubernetes.io/instance=connaisseur" and that's why it's not deleted by the first command.

**Reply by xopham** [reply]
You are correct! It used to not be labelled to allow people naming themselves since the crds cannot be rendered flexibly via Helm. I therefore mentioned that in the [release announcement](https://github.com/sse-secure-systems/connaisseur/discussions/187). However, we figured that the desire of renaming the instance is probably lower than having a clean uninstall and therefore changed that with version 2.0 in #185
